slh-dsa/src/hashes.rs

```diff
@@ -7,32 +7,43 @@ mod shake;
 
 use core::fmt::Debug;
 
+use digest::Update;
 use hybrid_array::{Array, ArraySize};
+use signature::Result;
 
 pub use sha2::*;
 pub use shake::*;
 
 use crate::{PkSeed, SkPrf, SkSeed, address::Address};
 
+/// `Digest` parameter for [`DigestSigner`] and [`DigestVerifier`].
+///
+/// [`DigestSigner`]: signature::DigestSigner
+/// [`DigestVerifier`]: signature::DigestVerifier
+pub trait HashDigest {
+    /// The associated `Digest` type.
+    type Digest: Update;
+}
```

Comment:
I originally had `type Digest` on `HashSuite` directly, but that made it publicly not accessible.

So I introduced `HashDigest` just to make it publicly accessible if users ever need to access the type.